runelite-client/src/main/java/net/runelite/client/plugins/inventorytags/InventoryTagsPlugin.java

```diff
@@ -95,25 +98,37 @@ InventoryTagsConfig provideConfig(ConfigManager configManager)
 		return configManager.getConfig(InventoryTagsConfig.class);
 	}
 
+	String getTag(int itemId)
+	{
+		String tag = configManager.getConfiguration(InventoryTagsConfig.GROUP, ITEM_KEY_PREFIX + itemId);
+		if (tag == null || tag.isEmpty())
+		{
+			return null;
+		}
+
+		return tag;
+	}
+
+	void setTag(int itemId, String tag)
+	{
+		configManager.setConfiguration(InventoryTagsConfig.GROUP, ITEM_KEY_PREFIX + itemId, tag);
+	}
+
+	void unsetTag(int itemId)
+	{
+		configManager.unsetConfiguration(InventoryTagsConfig.GROUP, ITEM_KEY_PREFIX + itemId);
+	}
+
 	@Override
 	protected void startUp() throws Exception
 	{
 		overlayManager.add(overlay);
-		tags = GSON.fromJson(config.getTaggedItems(), new TypeToken<Map<Integer, String>>()
-		{
-		}.getType());
-
-		if (tags == null)
-		{
-			tags = new HashMap<>();
-		}
 	}
 
 	@Override
 	protected void shutDown() throws Exception
 	{
 		overlayManager.remove(overlay);
-		tags.clear();
```

Comment:
state should be completely reset here, so hasTaggedItems and editorMode should be set to false